R/asset.R

```diff
@@ -0,0 +1,108 @@
+#' Add an external JavaScript asset as a dygraph dependency
+#'
+#' @param dygraph Dygraph to add asset to
+#' @param type Type of asset (valid options are: 'plugins', 'plotter' or 'dataHandler')
+#' @param name Name of asset
+#' @param path Path to asset JavaScript file
+#' @param version Asset version (e.g. version of package which provides the asset)
+#'
+#' @return A dygraph with the specified asset added.
+#'
+#' @importFrom htmltools htmlDependency
+#'
+#' @export
+dyAsset <- function(dygraph, type, name, path, version = "1.0") {
+  # create an html dependency for the asset js file
+  path <- normalizePath(path)
+  assetDependency <- htmlDependency(depName(type, name),
+                                    version,
+                                    src = dirname(path),
+                                    script = basename(path),
+                                    all_files = FALSE)
+
+  # add the asset javascript to the dependencies
+  if (is.null(dygraph$dependencies)) {
+    dygraph$dependencies <- list()
+  }
+  dygraph$dependencies[[length(dygraph$dependencies) + 1]] <- assetDependency
+
+  # return dygraph
+  dygraph
+}
+
+#' Make up a htmlDependency name based on asset type and name
+#'
+#' @param type Type of asset (valid options are: 'plugins', 'plotter' or 'dataHandler')
+#' @param name Name of asset
+#'
+#' @return A htmlDependency name.
+depName <- function(type, name) {
```

Comment:
We can remove the roxygen-style comments here (they result in the creation of a spurious .Rd file, not necessary since this is rightly a private function)